mkosi/__init__.py

```diff
@@ -659,13 +659,19 @@ def install_extra_trees(state: MkosiState) -> None:
         return
 
     with complete_step("Copying in extra file trees…"):
-        for tree in state.config.extra_trees:
-            if tree.is_dir():
-                copy_path(tree, state.root, preserve_owner=False)
+        for source, target in state.config.extra_trees:
+            t = state.root
+            if target:
+                t = state.root / str(target).lstrip("/")
```

Comment:
```suggestion
                t = state.root / target.relative_to("/")
```
This has the benefit of not round-tripping it through a string and also breaking for relative paths, thus making it harder to escape to path above `state.root`, well, at least I think of that as a benefit.